pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1412,6 +1421,12 @@ func TestBindPodVolumes(t *testing.T) {
 			}()
 		}
 
+		// This is used for checking the node affinity on bound volumes
+		if scenario.updateProvisionedPVCs {
+			pvc := pvcSetSelectedNode(scenario.cachedPVC, pod.Spec.NodeName)
```

Comment:
I'm not sure how this test case is validating PV.NodeAffinity since it only modifies the PVC.  Also, would prefer this test case to be implemented through delayFunc

Reply:
> Also, would prefer this test case to be implemented through delayFunc

Done.

> I'm not sure how this test case is validating PV.NodeAffinity since it only modifies the PVC.

AFAIU, the PVs are taken from the `pvAssumeCache` object. This is only necessary because `volumeBinder.checkBoundClaims()` takes in a list of PVCs, but it fetches the PVs referenced in these PVCs from `volumeBinder.pvCache` (which already contains the PVs).